modules/indexer/code/indexer.go

```diff
@@ -150,10 +153,14 @@ func Init() {
 
 				if err := index(ctx, indexer, indexerData.RepoID); err != nil {
 					log.Error("index: %v", err)
-					continue
+					if indexer.Ping() {
+						continue
+					}
+					// Add back to queue
+					unhandled = append(unhandled, datum)
 				}
 			}
-			return nil
+			return unhandled
```

Comment:
But the order is changed?

Reply:
Order in batch is not changed and returned data is added in front of queue imho